python/mlc_chat/serve/entrypoints/openai_entrypoints.py

```diff
@@ -299,6 +386,9 @@ async def completion_stream_generator() -> AsyncGenerator[str, None]:
                     # Ignore empty delta text -- do not yield.
                     continue
 
+                if conv_template.use_function_calling:
+                    finish_reason = "tool_calls"
```

Comment:
would be good to cross check finish reason, latest OAI API seems to use `function_call` https://platform.openai.com/docs/guides/text-generation/chat-completions-api

Reply:
https://platform.openai.com/docs/api-reference/chat/object#chat/object-choices

Expand the `choices` field.